libs-scala/resources/src/test/suite/scala/com/digitalasset/resources/ResourceOwnerSpec.scala

```diff
@@ -734,6 +733,56 @@ final class ResourceOwnerSpec extends AsyncWordSpec with Matchers {
         )
       }
     }
+
+    "convert to a ResourceOwner, which ensures, that no TimerTask is running after the resource is released" in {
+      val finishLongTakingTask = Promise[Unit]()
+      val timerResource = Factories.forTimer(() => new Timer("test timer")).acquire()
+      for {
+        timer <- timerResource.asFuture
+        (timerReleased, extraTimerTaskStarted) = {
+          timer.schedule(
+            new TimerTask {
+              override def run(): Unit = {
+                Await.result(finishLongTakingTask.future, Duration(10, "seconds"))
+              }
+            },
+            0L,
+          )
+          info("As scheduled a long taking task")
+          val released = timerResource.release()
+          info("And as triggered release of the timer resource")
+          Thread.sleep(100)
+          info("And as waiting 100 millis")
+          released.isCompleted shouldBe false
+          info("The release should not be completed yet")
+          val extraTimerTaskStarted = Promise[Unit]()
```

Comment:
I think this could be just an atomic boolean as you just check for future completeness